src/annotator/guest.js

```diff
@@ -491,34 +552,59 @@ export default class Guest extends Delegator {
     const targets = Promise.all([info, selectors]).then(setTargets);
 
     targets.then(() => this.publish('beforeAnnotationCreated', [annotation]));
-    targets.then(() => this.anchor(annotation));
+    targets.then(() => this.anchor(/** @type {AnnotationData} */ (annotation)));
 
     if (!annotation.$highlight) {
       this.crossframe?.call('showSidebar');
     }
     return annotation;
   }
 
+  /**
+   * Create a new annotation with the `$highlight` flag set.
+   *
+   * This flag indicates that the sidebar should save the new annotation
+   * immediately rather than prompting the user to write a comment.
```

Comment:
There are kind of two things that happen when a highlight is created:

1. Automatically saved (you've captured that)
2. The sidebar does not go into an editing mode for the annotation-highlight in question

You've obliquely captured number 2 by referencing "write a comment," but perhaps being more explicit about the UI "mode" could be useful?